contrib/go.mod

```diff
@@ -0,0 +1,23 @@
+module github.com/envoyproxy/go-control-plane/contrib
+
+go 1.20
```

Comment:
JFYI, we should be prepared for some pushback on this. It's possible that some consumer of this module will be using earlier Go versions (see #271 for example).

Reply:
I can try and roll back to 1.18 but that is a bare minimum requirement for workspaces

Reply:
+1 please don't use 1.20.  In grpc-go, which uses this, we support 1.18+ and also haven't made any changes AFAIK that would prevent even 1.17 or maybe 1.16 from working (e.g. no use of `any` or generics).  IMO this (and the use of new language features) should stay at the oldest version you can reasonably handle supporting.

Reply:
@alecholmez it sounds like this is the most contentious point within this change set.